lightclient/src/background.rs

```diff
@@ -287,22 +319,37 @@ impl<TPlatform: PlatformRef, TChain> BackgroundTask<TPlatform, TChain> {
                     return;
                 };
 
-                if let Some(sender) = self.subscriptions.get_mut(&(id, chain_id)) {
-                    // Send the current notification response.
-                    if sender.send(result).is_err() {
-                        tracing::warn!(
-                            target: LOG_TARGET,
-                            "Cannot send notification to subscription id={id} chain={chain_id:?} method={method}",
-                        );
+                let Some(subscription_state) = self.subscriptions.get_mut(&(id, chain_id)) else {
+                    tracing::warn!(
+                        target: LOG_TARGET,
+                        "Subscription response id={id} chain={chain_id:?} method={method} is not tracked",
+                    );
+                    return;
+                };
+                if subscription_state.sender.send(result).is_ok() {
+                    // Nothing else to do, user is informed about the notification.
+                    return;
+                }
 
-                        // Remove the sender if the subscription dropped the receiver.
-                        self.subscriptions.remove(&(id, chain_id));
-                    }
-                } else {
+                // User dropped the receiver, unsubscribe from the method and remove internal tracking.
+                let Some(subscription_state) = self.subscriptions.remove(&(id, chain_id)) else {
```

Comment:
I'd have to check; is the _only_ kind of error that can be returned from the channel above one that indicates it was closed by the receiver?

Reply:
Yep, taken from https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.UnboundedSender.html#method.send:

> If the receive half of the channel is [close](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.UnboundedReceiver.html#method.close)d, either due to close being called or the [UnboundedReceiver](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.UnboundedReceiver.html) having been dropped, this function returns an error. The error includes the value passed to send.